File: src/main/java/io/gittul/domain/github/GitHubBatchService.java
package io.gittul.domain.github;

import io.gittul.domain.post.dto.PostFeedResponse;
import io.gittul.domain.user.UserInquiryService;
import io.gittul.domain.user.entity.User;
import io.gittul.infra.notify.SlackService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Slf4j
@Service
@RequiredArgsConstructor
public class GitHubBatchService {
    private final GithubService githubService;
    private final UserInquiryService userInquiryService;

    private final SlackService slackService;

    @Value("${gittul.bot.id}") // Todo. 토큰을쓸까
    private Long adminId;

    @Scheduled(cron = "0 30 0 * * ?")
    public void scheduleDailyTrendingRepositoriesSummery() {
        User admin = userInquiryService.getUserById(adminId);
        List<PostFeedResponse> responses = githubService.getDailyTrendingRepositoriesSummery(admin);

        log.info("[요약 생성] {} 개의 트렌딩 레포지토리 요약이 생성되었습니다.", responses.size());
        slackService.sendMessage("[오늘의 트랜딩 레포지토리 요약]", responses.stream()
                .collect(Collectors.toMap(PostFeedResponse::title, PostFeedResponse::description)));
    }
}

File: app/src/main/kotlin/io/gittul/app/domain/github/GitHubBatchService.kt
package io.gittul.app.domain.github

import io.gittul.app.domain.user.UserInquiryService
import io.gittul.app.global.logger
import io.gittul.app.infra.notify.SlackService
import io.gittul.core.global.exception.CustomException
import org.springframework.beans.factory.annotation.Value
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Service

@Service
class GitHubBatchService(
    private val githubService: GithubService,
    private val userInquiryService: UserInquiryService,
    private val slackService: SlackService
) {


    @Value("\${gittul.bot.id}") // Todo. 토큰을쓸까
    private val adminId: Long? = null

    @Scheduled(cron = "0 30 0 * * ?")
    fun scheduleDailyTrendingRepositoriesSummery() {
        val admin = userInquiryService.getUserById(adminId!!)
        val responses = githubService.getDailyTrendingRepositoriesSummery(admin)

        logger().info("[요약 생성] {} 개의 트렌딩 레포지토리 요약이 생성되었습니다.", responses.size)
        slackService.sendMessage(
            "[오늘의 트랜딩 레포지토리 요약]", responses.associate { it.title to it.description }
        )
    }
}
